src/solace_ai_connector/components/general/openai/openai_chat_model_base.py

```diff
@@ -155,37 +169,50 @@ def invoke_stream(self, client, message, messages):
         current_batch = ""
         first_chunk = True
 
-        for chunk in client.chat.completions.create(
-            messages=messages,
-            model=self.model,
-            temperature=self.temperature,
-            stream=True,
-        ):
-            if chunk.choices[0].delta.content is not None:
-                content = chunk.choices[0].delta.content
-                aggregate_result += content
-                current_batch += content
-                if len(current_batch.split()) >= self.stream_batch_size:
-                    if self.stream_to_flow:
-                        self.send_streaming_message(
-                            message,
-                            current_batch,
-                            aggregate_result,
-                            response_uuid,
-                            first_chunk,
-                            False,
-                        )
-                    elif self.stream_to_next_component:
-                        self.send_to_next_component(
-                            message,
-                            current_batch,
-                            aggregate_result,
-                            response_uuid,
-                            first_chunk,
-                            False,
-                        )
-                    current_batch = ""
-                    first_chunk = False
+        max_retries = 3
+        while max_retries > 0:
```

Comment:
If retry while streaming, what happens to the tokens that have already been streamed

Reply:
We can't call them back. The way this is written, once we get one chunk we won't retry anymore see line 181. The streamed tokens will be delivered to the requester and (right now) the request would have to timeout. Not the best behaviour, but it would be a big thing to fix now and it isn't any worse than what is already in place.